docs/tutorials/ai/durable-ai-agent/index.md

```diff
@@ -0,0 +1,6078 @@
+
+```python
+import json
+
+from tools.search_flights import search_flights
+
+if __name__ == "__main__":
+
+    flights = search_flights(
+        {
+            "origin": "ORD",
+            "destination": "DFW",
+            "dateDepart": "2025-09-20",
+            "dateReturn": "2025-09-22",
+        }
+    )
+    print(json.dumps(flights, indent=2))
+```
+
+This test searches for a flight from Chicago to Dallas-Fort Worth.
+However, since this tool can operate in either a mock mode or live API mode, there are two ways to verify it.
+
+#### Testing the mocked `search_flight` tool
+
+Let's start by testing it without the RapidAPI key. 
+If you have that set in your `.env` file, comment it out for now, or skip this step.
+
+Run the test using the following command:
+
+```command
+uv run scripts/search_flight_test.py
+```
+
+Your output will vary, as the mock data function randomly generates results.
+The output should, however, look something like this with more items in the results list:
+
+```output
+{
+  "currency": "USD",
+  "destination": "DFW",
+  "origin": "ORD",
+  "results": [
+    {
+      "operating_carrier": "Southwest Airlines",
+      "outbound_flight_code": "WN427",
+      "price": 462.43,
+      "return_flight_code": "WN744",
+      "return_operating_carrier": "Southwest Airlines"
+    }
+  ]
+}
+```
+
+If you aren't planning on using the Sky Scrapper API, you can skip this next step and continue if you'd like.
+
+#### Testing the Sky Scrapper powered `search_flights` tool
+
+Testing the API-powered version of the tool is similar to testing the mocked version.
+
+First, if you haven't uncommented the `RAPID_API` lines in your `.env` file and added your API key, do this before running the test.
+You will also need to uncomment the `RAPIDAPI_HOST_FLIGHTS` environment variable as this is the endpoint the tool will be accessing.
+
+```bash
+RAPIDAPI_KEY=YOUR_RAPID_API_KEY
+RAPIDAPI_HOST_FLIGHTS=sky-scrapper.p.rapidapi.com
+```
+
+Next, review the code in `scripts/search_flights_test.py` and make sure that the `dateDepart` and `dateReturn` dates are both in the future.
+At this point you have no way of determining if the dates are in the past, and the API will return an error if you try to search for flights in the past. 
+
+Once you've reviewed the code, run the test using the following command:
+
+```command
+uv run scripts/search_flight_test.py
+```
+
+If you've changed the dates or cities, you may see different results, but the format should be similar to this:
+
+```output
+Searching for: ORD
+Searching for: DFW
+{
+  "origin": "ORD",
+  "destination": "DFW",
+  "currency": "USD",
+  "results": [
+    {
+      "outbound_flight_code": "NK824",
+      "operating_carrier": "Spirit Airlines",
+      "return_flight_code": "NK828",
+      "return_operating_carrier": "Spirit Airlines",
+      "price": 119.98
+    },
+  ]
+}
+```
+
+:::warning
```

Comment:
This renders as "DANGER" which is I think a little too forceful here; this can be an `info` block instead I think.